src/parsers/xyz.js

```diff
@@ -33,16 +33,41 @@ function validateLine(xyzLine, index) {
     });
 }
 
+/**
+ * Function checks the number of atoms listed in the BasisText editor.
+ * If the number of atoms > Basis.nonPeriodicMaxAtomsCount, then an error is thrown.
+ *
+ * @param {String} xyzTxt
+ */
+export function validateNumberOfAtoms(xyzTxt) {
+    const xyzArray = xyzTxt.split(/\r?\n/);
+    const nAtoms = xyzArray.length;
+    if (nAtoms > Basis.nonPeriodicMaxAtomsCount) {
+        throw new Error("Non-Periodic max atom limit exceeded");
+    }
+}
+
 /**
  * Validates that passed string is well-formed XYZ file.
  * @param xyzTxt {String}
+ * @return {Number}
  */
 export function validate(xyzTxt) {
-    s(xyzTxt)
-        .trim()
-        .lines()
-        .filter((x) => x.trim() !== "")
-        .forEach(validateLine);
+    try {
+        s(xyzTxt)
+            .trim()
+            .lines()
+            .filter((x) => x.trim() !== "")
+            .forEach(validateLine);
```

Comment:
Let's rename previous `validate` to `validateFormat` and then use it in lines 57-61 inside `validateAll`